tools/pipelines/templates/build-npm-package.yml

```diff
@@ -1,598 +1,540 @@
+                    echo 'curl -u ":<REDACTED>"
+                    "https://dev.azure.com/fluidframework/internal/_apis/build/builds/$BUILD_ID/timeline"'
 
-        - ${{ if or(eq(parameters.publishDocs, true), eq(parameters.taskBuildDocs, true)) }}:
-          - task: PublishPipelineArtifact@1
-            displayName: Publish Artifact - _api-extractor-temp
-            inputs:
-              targetPath: '${{ parameters.buildDirectory }}/_api-extractor-temp'
-              artifactName: '_api-extractor-temp'
-              publishLocation: 'pipeline'
+                    curl -u ":$ADO_API_TOKEN"
+                    "https://dev.azure.com/fluidframework/internal/_apis/build/builds/$BUILD_ID/timeline\?api-version\=6.0-preview.1"
+                    > $(pipelineTelemetryWorkdir)/timingOutput/output.json
 
-        - ${{ if eq(parameters.packageManager, 'pnpm') }}:
-          # Reset the pnpm-lock.yaml file since it's been modified by the versioning. But for dependency caching we want
-          # the cache key (which is based on the contents of the lockfile) to be the unmodified file. So we reset the
-          # lockfile as the last step so that when the dependency cache is uploaded, the cache key matches what it was
-          # at the beginning of the CI job.
-          - task: Bash@3
-            displayName: Reset lockfile
-            inputs:
-              targetType: 'inline'
-              workingDirectory: ${{ parameters.buildDirectory }}
-              script: |
-                git checkout HEAD -- pnpm-lock.yaml
+                    pwd;
 
-          # Prune the pnpm store before it's cached. This removes any deps that are not used by the current build.
-          - task: Bash@3
-            displayName: Prune pnpm store
-            inputs:
-              targetType: 'inline'
-              workingDirectory: ${{ parameters.buildDirectory }}
-              script: |
-                pnpm store prune
-
-
-        - task: Bash@3
-          displayName: Check for extraneous modified files
-          inputs:
-            targetType: 'inline'
-            script: |
-              git status | grep -v -E 'package.json|package-lock.json|packageVersion.ts|lerna.json|.npmrc|build-tools/.npmrc|\(use.*' | grep '^\s' > git_status.log
-              if [ `cat git_status.log | wc -l` != "0" ]; then
-                cat git_status.log
-                echo "##vso[task.logissue type=error]Build should not create extraneous files"
-                exit -1;
-              fi
-
-        # This additional build step is used to run step not part of the main build. In build client pipeline,
-        # this is used to inject telemetry key file and upload built devtools extension.
-        - ${{ parameters.additionalBuildSteps }}
-
-      # Job - Component detection
-      - ${{ if eq(variables.componentDetection, true) }}:
-        - job: CG
-          displayName: Component Detection
-          pool: ${{ parameters.poolCG }}
```

Comment:
Ok I stand corrected, we do have instances where we override the pool for a job to be potentially different from the pool for the stage that contains the job. `poolCG` defaults to Small. We should try to find a way to maintain this behavior and have this job execute in a different pool. Otherwise, the `poolCG` parameter is being left without any uses and could go away.